packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,243 @@
+}
+
+function getColorName(name: string) {
+    const variantToUse = `option${getColorIndex(name, 8) + 1}`;
+
+    return `decorative-${variantToUse}`;
+}
+
+function AvatarInitials(props: AvatarInitialsProps) {
+    const { 
+        name,
+        size: sizeValue
+    } = props;
+    
+    const size = useResponsiveValue(sizeValue) ?? "md";
+
+    const initials = useMemo(() => {
+        const cleanName = name.replace(/\s+/g, " ").trim();
+        const [firstName, lastName] = cleanName.split(" ");
+        const letters = firstName && lastName
+            ? `${firstName.charAt(0)}${lastName.charAt(0)}`
+            : firstName.charAt(0);
+
+        return size === "xs" && letters.length > 1 ? letters.charAt(0) : letters;
+    }, [name, size]);
+
+    return (
+        <Text
+            size={AvatarToTextSizeAdapter[size]}
+            className={styles["hop-Avatar__initials"]}
+        >
+            {initials}
+        </Text>
+    );
+}
+
+function Avatar(props: AvatarProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, AvatarContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        "aria-label": ariaLabel,
+        className,
+        fallbackSrc,
+        imageProps,
+        isDisabled,
+        name,
+        size: sizeValue,
+        slot,
+        src,
+        style,
+        ...otherProps
+    } = ownProps;
+    const domProps = filterDOMProps(otherProps);
+    
+    const { onError, onLoad, ...otherImageProps } = imageProps ?? {};
+    const [imageUrl, status] = useImageFallback({ src, fallbackSrc, onError, onLoad });
+    const imageLoaded = status === "loaded";
+    const imageFailed = status === "failed";
+
+    const size = useResponsiveValue(sizeValue) ?? "md";
+    const isBrokenImage = src && imageFailed && fallbackSrc !== null;
+    const isImage = src && !imageFailed;
+    const isInitials = !src || (src && imageFailed && fallbackSrc === null);
+
+    const classNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            size,
+            isBrokenImage && "broken-image",
+            isImage && "image",
+            isInitials && getColorName(name)
+        ),
+        stylingProps.className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...stylingProps.style,
+        ...style
+    };
+
+    const handleError: ReactEventHandler<HTMLImageElement> = e => {
+        onError?.(e);
+    };
+
+    let content = <AvatarInitials
+        {...otherProps}
+        name={name}
+        size={size}
+    />;
+    
+    if (imageLoaded) {
+        content = <img
+            {...filterDOMProps(otherImageProps)}
+            alt=""
+            aria-hidden
+            className={styles["hop-Avatar__image"]}
+            onError={handleError}
+            src={imageUrl}
+        />;
+    }
+
+    if (imageFailed) {
+        if (fallbackSrc !== null) {
```

Comment:
i feel like the 2 return should happen before we declare the "content" variable. content is only usefull if we pass all those checks